### PR TITLE
[MIG] membership_variable_period: Migration to 12.0

### PR DESCRIPTION
**Pull request by chienandalu**
cc @Tecnativa

### CODE REVIEW

**Review by Tardo — APPROVED** [review]
LGTM, code review

File: membership_variable_period/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    membership_interval_unit = fields.Selection(
+        selection=[('days', 'days'),
+                   ('weeks', 'weeks'),
+                   ('months', 'months'),
+                   ('years', 'years')],
+        string="Interval unit", default='years')
+
+    def _correct_vals_membership_type(self, vals, membership_type):
+        if membership_type == 'variable':
+            vals['membership_date_from'] = False
```
Comment (nikul-serpentcs) [inline membership_variable_period/models/product_template.py]: @chienandalu IMO vals['membership_date_from'] = vals['membership_date_to'] = False

**Review by nikul-serpentcs — COMMENTED** [review]
minor change

File: membership_variable_period/models/product_template.py (hunk 1 above)
Comment (chienandalu) [reply membership_variable_period/models/product_template.py]: Ok, it's almost a matter of taste :)  but I changed to the oneliner way

**Comment by nikul-serpentcs** [comment]
@chienandalu Fix Transifex commit's `OCA Transbot updated translations from Transifex`

**Comment by chienandalu** [comment]
> @chienandalu Fix Transifex commit's `OCA Transbot updated translations from Transifex`

@nikul-serpentcs That's the history of the module. What do you want me to fix there?

**Comment by nikul-serpentcs** [comment]
>@nikul-serpentcs That's the history of the module. What do you want me to fix there?


Because It's not needed anywhere, Only Increase the number of Commit's

**Comment by chienandalu** [comment]
> Because It's not needed anywhere, Only Increase the number of Commit's

We only squash transifex consecutive commits (or event author/logical commits). Otherwise we let history as it is

**Comment by pedrobaeza** [comment]
Yes, if you see my usual text is:

> Please squash together **adjacent** "OCA Transbot.." commits using https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

so I see this one correct.

**Review by rafaelbn — APPROVED** [review]
tested in runbot

**Pull request merged**